pkg/inventory/web/handler.go

```diff
@@ -266,10 +259,40 @@ func (r *WatchWriter) Error(err error) {
 //
 // An event watch has ended.
 func (r *WatchWriter) End() {
-	r.log.V(4).Info("event: ended.")
+	r.log.V(3).Info("event: ended.")
 	r.send(model.Event{
 		Action: model.End,
 	})
+	r.done = true
+	time.Sleep(50 * time.Millisecond)
+	_ = r.webSocket.Close()
+}
+
+//
+// Write event to the socket.
+func (r *WatchWriter) send(e model.Event) {
+	if r.done {
+		return
+	}
+	event := Event{
+		ID:     e.ID,
+		Action: e.Action,
+	}
+	if e.Model != nil {
+		event.Resource = r.builder(e.Model)
+	}
+	if e.Updated != nil {
+		event.Updated = r.builder(e.Updated)
+	}
+	err := r.webSocket.WriteJSON(event)
+	if err != nil {
+		r.log.V(4).Error(err, "websocket send failed.")
+	}
+
+	r.log.V(5).Info(
+		"event sent.",
+		"event",
+		event)
```

Comment:
The send() just moved to after the public (exported) methods.
A healthy pattern, IMHO.